### DISCUSSION TITLE
ogre3d

### DISCUSSION THREAD

**Discussion by revelator** [msys2/MINGW-packages#8214] [General]
Been working on updating the ogre3d version to latest release but as it turned out a few not so nice caveats are present.
d3d is utterly broken this actually turned out to be a bug in the source code and a possible fix was posted unfortunatly it does not work even with these, so only opengl and opengl+3 renderers are working atm.
imgui is no longer nessesary to download seperatly and no longer works in the old package due to changes upstream.

I was wondering if i should do a pull request with so many broken features or wait for the next release where they have hopefully ironed out the bugs ?.

**Comment by Biswa96** [comment]
I'd go with next release and fix the code in upstream if necessary.

**Comment by revelator** [comment]
well turned out the d3d problems were caused by swig of all things :( after a rebuild adding the absolute paths to the different interpreters it uses (well those that it can use on windows anyway) it started to work. Still having to use upstream patches ment for the next release to fix something in the current release does not quite sit well with me.

**Reply by revelator** [reply]
finally working again with version 1.12.12, some patches can be removed -> 011-mingw-check-security-coockie-not-available.patch 012-add-link-library.patch and one was added 011-irrxml-mingw.patch. Also no longer nessesary with the imgui hack.
